graphql/exec.go

```diff
@@ -51,23 +54,25 @@ func collectFields(doc *ast.QueryDocument, selSet ast.SelectionSet, satisfies []
 
 			f.Selections = append(f.Selections, sel.SelectionSet...)
 		case *ast.InlineFragment:
-			if !instanceOf(sel.TypeCondition, satisfies) {
+			if !shouldIncludeNode(sel.Directives, reqCtx.Variables) || !instanceOf(sel.TypeCondition, satisfies) {
```

Comment:
I might be wrong, but when I did this, `reqCtx` had no default arguments (ones from document). So this query
```graphql
query MyAwesomeQuery($shouldIncludeField: Boolean = false) {
    resolver @include(if: $shouldIncludeField) {
        __typename
    }
}
```
would still include in case no variables were explicitly passed. That's why I did this:
[https://github.com/vektah/gqlgen/pull/187/files#diff-47a8f73d5b7f0dbad02ee4611bbedb10R79](https://github.com/vektah/gqlgen/pull/187/files#diff-47a8f73d5b7f0dbad02ee4611bbedb10R79)
BUT again, I might be very-very wrong.

Reply:
:+1: it's probably worth a testcase